### PR TITLE
Added check for google sql contained db authentication

### PR DESCRIPTION
**Pull request by liamg**

### CODE REVIEW

File: internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"github.com/aquasecurity/tfsec/internal/app/tfsec/scanner"
+)
+
+func init() {
+	scanner.RegisterCheckRule(rule.Rule{
+		Service:   "sql",
+		ShortCode: "no-contained-db-auth",
+		Documentation: rule.RuleDocumentation{
+			Summary: "Contained database authentication should be disabled",
+			Explanation: `	`,
```
Comment (owenrumney) [inline internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go:27]: No explanation

File: internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	region           = "us-central1"
+	settings {
+	    database_flags {
+		    name  = "contained database authentication"
+		    value = "off"
+		}
+	}
+}
+			`,
+			Links: []string{},
```
Comment (owenrumney) [inline internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go:50]: no links

File: internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go (hunk 1 above)
Comment (liamg) [reply internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go]: :+1:

File: internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go (hunk 2 above)
Comment (liamg) [reply internal/app/tfsec/rules/google/sql/no_contained_db_auth_rule.go]: :+1:

**Review by owenrumney — APPROVED** [review]

**Pull request merged**